modules/calib3d/include/opencv2/calib3d.hpp

```diff
@@ -1725,14 +1726,16 @@ and/or @ref CALIB_FIX_ASPECT_RATIO, @ref CALIB_FIX_PRINCIPAL_POINT or @ref CALIB
 are specified, some or all of fx, fy, cx, cy must be initialized before calling the function.
 @param distCoeffs Input/output vector of distortion coefficients
 \f$\distcoeffs\f$.
-@param rvecs Output vector of rotation vectors (@ref Rodrigues ) estimated for each pattern view
-(e.g. std::vector<cv::Mat>>). That is, each i-th rotation vector together with the corresponding
-i-th translation vector (see the next output parameter description) brings the calibration pattern
-from the object coordinate space (in which object points are specified) to the camera coordinate
-space. In more technical terms, the tuple of the i-th rotation and translation vector performs
-a change of basis from object coordinate space to camera coordinate space. Due to its duality, this
-tuple is equivalent to the position of the calibration pattern with respect to the camera coordinate
-space.
+@param rvecs Output vector of 3x3 rotation matrices or 3x1 rotation vectors (@ref Rodrigues )
+estimated for each pattern view (e.g. std::vector<cv::Mat>>). That is, each i-th rotation matrix
+together with the corresponding i-th translation vector (see the next output parameter description)
+brings the calibration pattern from the object coordinate space (in which object points are
+specified) to the camera coordinate space. In more technical terms, the tuple of the i-th rotation
+and translation vector performs a change of basis from object coordinate space to camera coordinate
+space. Due to its duality, this tuple is equivalent to the position of the calibration pattern with
+respect to the camera coordinate space.  If @ref CALIB_FIX_EXTRINSIC is specified, rotation
+matrices and translation vectors equal in size to the number of images must be initialized and will
+be used for calibration.
```

Comment:
> will be used for calibration

This contradicts with `OutputArrayOfArrays rvecs` parameter declaration which is critical for bindings generators (Python/Java/etc code).

Perhaps we should make dedicated overload (at least in public headers).

Reply:
Thanks @alalek .  I see 4 api versions of calibrate camera in calib3d.hpp:
- calibrateCameraExtended
- calibrateCamera
- calibrateCameraROExtended
- calibrateCameraRO

Here's an initial proposal for the overload:
- calibrateCameraFEExtended:  Based on calibrateCameraExtended but with rvecs and tvecs as InputOutputArrayOfArrays
- calibrateCameraFE:  Based on calibrateCamera but with rvecs and tvecs as InputOutputArrayOfArrays

The "FE" stands for "FixedExtrinsics".  This is similar to how calibrateCameraROExtended and calibrateCameraRO are declared and how "RO" stands for "ReleasingObject".  I selected InputOutputArrayOfArrays instead of InputArrayOfArrays  in case one day we want the user to be able to pass in an initial guess for extrinsics that will be optimized by the LM solver.

I'm not sure whether want to the method of releasing object to be combined with fixed extrinsics so I didn't base an overload from those methods.

What's your opinion?

Reply:
Thanks @alalek .  I see 4 api versions of calibrate camera in calib3d.hpp:
- calibrateCameraExtended
- calibrateCamera
- calibrateCameraROExtended
- calibrateCameraRO

Here's an initial proposal for the overload:
- calibrateCameraFEExtended:  Based on calibrateCameraROExtended but with rvecs and tvecs as InputOutputArrayOfArrays
- calibrateCameraFE:  Based on calibrateCameraRO but with rvecs and tvecs as InputOutputArrayOfArrays

The "FE" stands for "FixedExtrinsics".  This is similar to how calibrateCameraROExtended and calibrateCameraRO are declared and how "RO" stands for "ReleasingObject".  I selected InputOutputArrayOfArrays instead of InputArrayOfArrays  in case one day we want the user to be able to pass in an initial guess for extrinsics that will be optimized by the LM solver.

Basing the extension on calibrateCameraRO also allows the method of releasing object to be combined with fixed extrinsics.

What's your opinion?